_docs/0.3/usage/dataset-comparison.md

```diff
@@ -0,0 +1,135 @@
+
+Now how to use it all together:
+
+- `--format`, `--new-format`, `--ref-format` for choosing the format type. This can be `parquet`, `csv`, `xml`, `jdbc`, etc.
+- `--ref-path`, `--new-path` and `--out-path` define paths on HDFS of reference data path, newly created data path and diff output path, repspectively. In case of usage of `jdbc` format these can be replaced by `--ref-dbtable` and `--ned-dbtable`
+- `--keys` to specifie the primary key(s) of the table, e.g. `--keys ID1, ID2`. These can be only top level columns.
+- format specific keys prefixed with `--`, so `--header`, `--rowTag`, `--user`, etc.
+
+**Mandatory fields** are logically the paths/dbtables, format (at least the general one) and any format specific keys that are prescribed by the format used.
+
+**NOTE** if you want to use `jdbc` format or others, requiring specific drivers, you should have appropriate drivers on a class path.
+
+
+#### Usage as library
+
+For usage as library use class `za.co.absa.hermes.datasetComparison.DatasetComparator`. 
+
+**Prescription of DatasetComparator class**:
+
+```scala
+/**
+ * Class that is the brain of the DatasetComparison module. This class should be used in case of using DatasetComparison
+ * as a library. In case of running the DatasetComparison as SparkJob, please use the DatasetComparisonJob.
+ *
+ * @param dataFrameReference is a Dataframe used in a comparison as the origin of truth
+ * @param dataFrameActual is a Dataframe that is being tested
+ * @param keys is a set of primary keys of the dataset. This highly increases accuracy of the output as we are then able
+ *             to pinpoint the differences
+ * @param config Config object holding project based configurable parameters. Difference to the cliOptions is that these
+ *               are meant to stay the same for the project, while cliOptions change for each test
+ * @param optionalSchema Optional schema to cherry-pick columns form the two dataframes to compare. For example, if you
+ *                       have a timestamp column that will never be the same, you provide a schema without that timestamp
+ *                       and it will not be compared.
+ * @param sparkSession Implicit spark session.
+ */
+class DatasetComparator(dataFrameReference: DataFrame,
+                        dataFrameActual: DataFrame,
+                        keys: Set[String] = Set.empty[String],
+                        config: DatasetComparisonConfig = new TypesafeConfig(None),
+                        optionalSchema: Option[StructType] = None)
+                       (implicit sparkSession: SparkSession)
+```
+
+You create an instance of `DatasetComparator` and then invoke parametrless `compare` method. This method returns `ComparisonResult`.
```

Comment:
Again personal vs. imperative mood. The personal is getting in balance with imperative as I go further in the text. 
So up to you, but I would stick to one - whichever you prefer - and keep to it.